doc/development/deprecations.rst

```diff
@@ -66,6 +66,12 @@ Pending deprecations
   - Deprecated in v0.32
   - Will be removed in v0.33
 
+* The ``QuantumScript.set_parameters`` method has been deprecated. Please use
+  ``QuantumScript.bind_new_parameters`` instead.
```

Comment:
we should put the `tape.data` deprecation here too. I'd put it in the same entry since it's an alias for the same thing, but I'm not picky on that